services/worker-manager/src/api.js

```diff
@@ -609,8 +609,13 @@ builder.declare({
   }
 
   let expires, workerConfig;
+  const secret = `${slug.nice()}${slug.nice()}`;
   try {
-    const reg = await provider.registerWorker({worker, workerPool, workerIdentityProof});
+    const encryptedSecret = this.db.encrypt({ value: Buffer.from(secret, 'utf8') });
+    await worker.update(this.db, worker => {
+      worker.secret = encryptedSecret;
+    });
```

Comment:
This isn't a huge issue but doesn't this allow me to write whatever I want into the secret column of any worker I want since we haven't verified the identity of the client yet at this point? I think doing this after `provider.registerWorker` makes more sense?

Reply:
Agreed. Makes more sense.